### PR TITLE
Upgrade Istio resource to 1.10.0 version

### LINKED ISSUE #11423: Upgrade Istio to 1.10
**Description**

Upgrade Istio to 1.10

AC:

- istio running on 1.10.0 at least
- updated istiooperator resources
- security scans green (?)
- release notes
- all sidecars runs on 1.10.0 (include improvements in proxy-reset-job)

**Reasons**

For security & support reasons we should be up to date with the latest Istio release.

Attachments

https://istio.io/latest/news/releases/1.10.x/announcing-1.10/change-notes/
https://istio.io/latest/news/releases/1.10.x/announcing-1.10/upgrade-notes/

### PR DESCRIPTION
**Pull request by cnvergence**
**Description**
For security & support reasons we should be up to date with the latest Istio release.
https://istio.io/latest/news/releases/1.10.x/announcing-1.10/change-notes/
https://istio.io/latest/news/releases/1.10.x/announcing-1.10/upgrade-notes/

Changes proposed in this pull request:

- Upgrade Istio to 1.10.0 version.
- Remove deprecated global.arch settings in favour of the Kubernetes affinity rules.
- Change the services that bind to the _lo_ interface, to bind to all interfaces.
- Temporary use of legacy selectors option, as the follow up we will tweak our istio scripts to align with the new selector logic.

From 1.10 Istio changed the networking behaviour for the inbound traffic, the proxy no longer redirects the traffic to the _lo_ interface and forwards it to the application on _eth0_.
For more info, please refer to the docs [#1](https://istio.io/latest/blog/2021/upcoming-networking-changes) and [#2](https://istio.io/latest/news/releases/1.10.x/announcing-1.10/upgrade-notes/#inbound-forwarding-configuration).

**Related issue(s)**

Fixes #11423

### CODE REVIEW

File: resources/istio/files/istio-operator-cluster-evaluation.yaml [hunk 1]
```diff
@@ -301,6 +301,7 @@ spec:
       replicaCount: 1
       traceSampling: 1
     sidecarInjectorWebhook:
+      useLegacySelectors: true
```
Comment (piotrkpc) [inline resources/istio/files/istio-operator-cluster-evaluation.yaml:304]: shouldn't we use https://istio.io/latest/blog/2021/discovery-selectors/#discovery-selectors-in-action instead?

File: resources/istio/files/istio-operator-cluster-production.yaml [hunk 2]
```diff
@@ -303,6 +299,7 @@ spec:
       replicaCount: 1
       traceSampling: 1
     sidecarInjectorWebhook:
+      useLegacySelectors: true
```
Comment (piotrkpc) [inline resources/istio/files/istio-operator-cluster-production.yaml:302]: shouldn't we use https://istio.io/latest/blog/2021/discovery-selectors/#discovery-selectors-in-action instead?

File: resources/istio/files/istio-operator-cluster.yaml [hunk 3]
```diff
@@ -300,6 +296,7 @@ spec:
       replicaCount: 1
       traceSampling: 1
     sidecarInjectorWebhook:
+      useLegacySelectors: true
```
Comment (piotrkpc) [inline resources/istio/files/istio-operator-cluster.yaml:299]: shouldn't we use https://istio.io/latest/blog/2021/discovery-selectors/#discovery-selectors-in-action instead?

File: resources/istio/files/istio-operator-minikube-evaluation.yaml [hunk 4]
```diff
@@ -312,6 +308,7 @@ spec:
       replicaCount: 1
       traceSampling: 1
     sidecarInjectorWebhook:
+      useLegacySelectors: true
```
Comment (piotrkpc) [inline resources/istio/files/istio-operator-minikube-evaluation.yaml:311]: shouldn't we use https://istio.io/latest/blog/2021/discovery-selectors/#discovery-selectors-in-action instead?

File: resources/istio/files/istio-operator-minikube.yaml [hunk 5]
```diff
@@ -310,6 +306,7 @@ spec:
       replicaCount: 1
       traceSampling: 1
     sidecarInjectorWebhook:
+      useLegacySelectors: true
```
Comment (piotrkpc) [inline resources/istio/files/istio-operator-minikube.yaml:309]: shouldn't we use https://istio.io/latest/blog/2021/discovery-selectors/#discovery-selectors-in-action instead?

**Review by piotrkpc — CHANGES_REQUESTED** [review]
I was reading some docs and if I understand this correctly, discovery selectors is the new selector work flow for injecting sidecar. I wonder if we can use it because the legacy selector logic will be removed.

https://docs.google.com/document/d/1umIDCxvTwfOBtNh-bbRb3eyOxJ_4jsSBXl4pfcAYqR0/edit?resourcekey=0-OOCyBAXemTcnpyDRjU48pQ#heading=h.xw1gqgyqs5b
https://istio.io/latest/blog/2021/discovery-selectors/

And if I understand all this correctly, after the upgrade and after we'll use new selector logic, docs below are no longer true.
https://github.com/kyma-project/kyma/blob/dbdd9427d86b262468b9ea3f254e0447dd6d479f/docs/service-mesh/03-03-sidecar-proxy-injection.md#L6-L13

File: resources/istio/files/istio-operator-minikube.yaml (hunk 5 above)
Comment (piotrkpc) [reply resources/istio/files/istio-operator-minikube.yaml:309]: during daily we have agreed to leave it for now and try to remove it in followup task.

File: resources/istio/files/istio-operator-minikube-evaluation.yaml (hunk 4 above)
Comment (piotrkpc) [reply resources/istio/files/istio-operator-minikube-evaluation.yaml:311]: during daily we have agreed to leave it for now and try to remove it in followup task.

File: resources/istio/files/istio-operator-cluster.yaml (hunk 3 above)
Comment (piotrkpc) [reply resources/istio/files/istio-operator-cluster.yaml:299]: during daily we have agreed to leave it for now and try to remove it in followup task.

File: resources/istio/files/istio-operator-cluster-production.yaml (hunk 2 above)
Comment (piotrkpc) [reply resources/istio/files/istio-operator-cluster-production.yaml:302]: during daily we have agreed to leave it for now and try to remove it in followup task.

File: resources/istio/files/istio-operator-cluster-evaluation.yaml (hunk 1 above)
Comment (piotrkpc) [reply resources/istio/files/istio-operator-cluster-evaluation.yaml:304]: during daily we have agreed to leave it for now and try to remove it in followup task.

**Comment by piotrkpc** [comment]
during daily we have agreed to leave it for now and try to remove it in followup task.

**Review by piotrkpc — COMMENTED** [review]
during daily we have agreed to leave it for now and try to remove it in followup task. LGTM

**Review by piotrkpc — APPROVED** [review]
/lgtm

**Review by werdes72 — APPROVED** [review]
/meow

**Pull request merged**